includes/Modules/Analytics.php

```diff
@@ -1328,28 +1322,67 @@ private function get_provisioning_redirect_uri() {
 	 * @since 1.8.0 Simplified to return a boolean and require account ID.
 	 *
 	 * @param string $property_id Property found in the existing tag.
-	 * @param string $account_id  Account ID the property belongs to.
-	 * @return bool True if the user has access, false otherwise.
+	 * @return array A string representing the accountID and a boolean representing if the user has access to the property.
 	 */
-	protected function has_access_to_property( $property_id, $account_id ) {
-		if ( empty( $property_id ) || empty( $account_id ) ) {
-			return false;
+	protected function has_access_to_property( $property_id ) {
+		if ( empty( $property_id ) ) {
+			return array(
+				'permission' => false,
+			);
 		}
 
+		$account_id = $this->parse_account_id( $property_id );
+
+		/**
+		 * Helper method to check check if a passed response
+		 * contains the property_id
+		 */
+		$has_property = function ( $response ) use ( $property_id ) {
+			if ( is_wp_error( $response ) ) {
+				return false;
+			}
+			foreach ( $response['properties'] as $property ) {
+				if ( $property->getId() === $property_id ) {
+					return true;
+				}
+			}
+			return false;
+		};
+
 		// Try to get properties for that account.
 		$properties = $this->get_data( 'properties-profiles', array( 'accountID' => $account_id ) );
-		if ( is_wp_error( $properties ) ) {
-			// No access to the account.
-			return false;
-		}
+		if ( ! is_wp_error( $properties ) ) {
+			// Ensure there is access to the property.
+			if ( $has_property( $properties ) ) {
+				return array(
+					'accountID'  => $account_id,
+					'permission' => true,
+				);
+			}
 
-		// Ensure there is access to the property.
-		foreach ( $properties['properties'] as $property ) {
-			if ( $property->getId() === $property_id ) {
-				return true;
+			// Check all of the accounts for this user.
+			$user_accounts_properties_profiles = $this->get_data( 'accounts-properties-profiles', array( 'existingAccountID' => $account_id ) );
+			if ( ! is_wp_error( $user_accounts_properties_profiles ) ) {
+				$user_account_ids = wp_list_pluck( $user_accounts_properties_profiles['accounts'], 'id' );
+				foreach ( $user_account_ids as $user_account_id ) {
+					// Skip the inferred account id, that ship has sailed.
+					if ( $account_id === $user_account_id ) {
+						continue;
+					}
+					$properties = $this->get_data( 'properties-profiles', array( 'accountID' => $user_account_id ) );
+					if ( $has_property( $properties ) ) {
+						return array(
+							'accountID'  => $account_id,
```

Comment:
This should be the same account ID that is being iterated with.
```suggestion
							'accountID'  => $user_account_id,
```